### ISSUE TITLE
Hybrid Feature Type Support

### ISSUE THREAD

**Issue by zhangpengshan** [ShifuML/shifu#405]
Now in Shifu only categorical and numerical features are supported. While considering such feature which is numerical:
1, 3, 0, A, V, C, A, 4, 2, ...

Consider such feature as Hybrid and compute binBoundary only in numerical part, binCategory in categorical part. Then it is good to do woe transform for NN/LR training.

**Comment by zhangpengshan** [comment]
1.	Define a new column type which is Hybrid compared with Categorical and Numerical

2.	Add a new config file for user to define hybrid feature which is hidden by default
User can define types like a,b,c or  <= 0 as expression

3.	For Hybrid feature: Do stats for both numerical stats to compute numerical boundaries and categorical stats to compute categories; Then such stats results will be set in both binBoundary and binCategory; ‘binPosRate,‘binCountWoe’ … will be combined by both. And Missing value is in last element if have.

4.	Norm for both WOE and ZScale
a)	WOE is easy, numerical and categorical value can both be set to woe values
b)	ZScale is OK, numerical values is on ZScore, categorical features can be set to binPosRate; at this part, as not meaningless for such norm, maybe better to limited for only woe transform

5.	Training & Eval
a)	NN/LR Training is OK as it is based on norm output
b)	Tree Training has some issues as in the before only numerical and categorical features can be used to split a node. As a hybrid feature, at first version, we can still treat it as numerical; then in the future, I am thinking to compute info gain on both numerical bins and categorical bins, then get split on both types. More computation will be excepted.

6.	PMML Support: http://dmg.org/pmml/v4-1/Functions.html#xsdElement_Apply 
After evaluating on PMML, DefinedFuntion and Apply can help us  to define such transform. I just implemented binary format NN model like GBT, in my tests of these days, I found the flexibility of binary format to define any transform and better SLA. In production, NN SLA is bad and 95% execution time is >= 25ms. While in may testing it is good as CAM GBT(1000 trees), which is 3ms in Horton eval jobs.
